linkschkr.go

```diff
@@ -0,0 +1,127 @@
+package linkschkr
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/html"
+)
+
+type Work struct {
+	site   string
+	result chan *Result
+}
+
+type Result struct {
+	site      string
+	up        bool
+	extraURIs []string
+	workerID  int
+}
+
+func Crawler(ID int, c <-chan *Work) {
+	for {
+		work := <-c
+		result := &Result{
+			site:     work.site,
+			up:       true,
+			workerID: ID,
+		}
+		var resp *http.Response
+		var err error
+		// Avoid to download binaries
+		switch {
+		case strings.HasSuffix(work.site, ".gz"):
+			resp, err = http.Head(work.site)
+		case strings.HasSuffix(work.site, ".bz2"):
+			resp, err = http.Head(work.site)
+		case strings.HasSuffix(work.site, ".msi"):
+			resp, err = http.Head(work.site)
+		case strings.HasSuffix(work.site, ".zip"):
+			resp, err = http.Head(work.site)
+		case strings.HasSuffix(work.site, ".pkg"):
+			resp, err = http.Head(work.site)
+		default:
+			resp, err = http.Get(work.site)
+		}
+		if err != nil || resp.StatusCode != 200 {
+			result.up = false
+			work.result <- result
+			continue
+		}
+
+		extraURIs := ParseHREF(resp.Body)
+		result.extraURIs = extraURIs
+		work.result <- result
+	}
+}
+
+func ParseHREF(r io.Reader) []string {
+	URIs := []string{}
+
+	doc, err := html.Parse(r)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var f func(*html.Node)
```

Comment:
A neat library to help with this is https://github.com/antchfx/htmlquery.

Reply:
I didn't like this one... it's harder to implement what I want... plus I cannot inject my own client to do the HTTP request.